packages/grafana-data/src/datetime/datemath.ts

```diff
@@ -40,12 +41,14 @@ export function isMathString(text: string | DateTime | Date): boolean {
  * @param text
  * @param roundUp See parseDateMath function.
  * @param timezone Only string 'utc' is acceptable here, for anything else, local timezone is used.
+ * @param now DateTimeInput to use as now. Useful when parsing multiple values and now needs to be the same. Without this, comparing results from subsequent parses is not guaranteed to be deterministic.
  */
 export function parse(
   text?: string | DateTime | Date | null,
   roundUp?: boolean,
   timezone?: TimeZone,
-  fiscalYearStartMonth?: number
+  fiscalYearStartMonth?: number,
+  now?: DateTimeInput
```

Comment:
I feel like at this point, the arg list is pretty horrible, considering it's 5 params all optional :D. At what point should we create a new function, with an options object? @ashharrison90 @joshhunt

Reply:
Yes, it’s quite unwieldy. Why is the text optional? 😂 

I think an overload is better on the balance of all things.

Reply:
The text to parse should be mandatory first parameter with optional options object.

Reply:
@aocenas @joshhunt I hear you. I've created a new function and marked this one as deprecated. Let me know what you think.